### PR TITLE
Add ability to reveal votes

### PR DESCRIPTION
**Pull request by ptare**
As with several of these React PR's, I'm basically improvising.

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: voter-dapp/src/ActiveRequests.js [hunk 1]
```diff
@@ -117,6 +122,23 @@ function ActiveRequests() {
   }, [subsequentFetchComplete, voteStatusesStringified, decryptionKeys, account]);
   const decryptionComplete = decryptedCommits && voteStatuses && decryptedCommits.length === voteStatuses.length;
 
+  const { send: batchRevealFunction, status } = useCacheSend("Voting", "batchReveal");
+  const onClickHandler = () => {
+    const reveals = [];
+    for (const index of Object.keys(checkboxesChecked)) {
```
Comment (mrice32) [inline voter-dapp/src/ActiveRequests.js]: nit: isn't this the same as `for (const index in checkboxesChecked)`?

File: voter-dapp/src/ActiveRequests.js [hunk 2]
```diff
@@ -117,6 +122,23 @@ function ActiveRequests() {
...
+        reveals.push({
+          identifier: pendingRequests[index].identifier,
+          time: pendingRequests[index].time,
+          price: decryptedCommits[index].price.toString(),
+          salt: web3.utils.hexToNumberString(decryptedCommits[index].salt)
+        });
+      }
+    }
+    batchRevealFunction(reveals);
+    setCheckboxesChecked({});
```
Comment (mrice32) [inline voter-dapp/src/ActiveRequests.js]: Just curious, do we not want to freeze the checkboxes until the result comes back? If we do, I think we should do that in a later PR, just wanted to hear your thoughts.

File: voter-dapp/src/ActiveRequests.js [hunk 3]
```diff
@@ -164,6 +194,7 @@ function ActiveRequests() {
         <TableHead>
           <TableRow>
             <TableCell>Price Feed</TableCell>
+            <TableCell>?</TableCell>
```
Comment (mrice32) [inline voter-dapp/src/ActiveRequests.js]: What is this?

File: voter-dapp/src/ActiveRequests.js [hunk 4]
```diff
@@ -136,25 +158,33 @@ function ActiveRequests() {
     );
   }
 
+  const hasPendingTransactions = status === "pending";
   const statusDetails = voteStatuses.map((voteStatus, index) => {
+    let currentVote = "";
+    if (voteStatus.committedValue && decryptedCommits[index].price) {
+      currentVote = web3.utils.fromWei(decryptedCommits[index].price.toString());
```
Comment (mrice32) [inline voter-dapp/src/ActiveRequests.js]: nit: should we use the `formatWei()` function from FormattingUtils.js? Or is there no reason to use it here? I forget exactly why we added that function.

File: voter-dapp/src/ActiveRequests.js [hunk 5]
```diff
@@ -13,9 +15,14 @@ import { decryptMessage, deriveKeyPairFromSignatureMetamask } from "./common/Cry
 const { getKeyGenMessage } = require("./common/EncryptionHelper.js");
 
 function ActiveRequests() {
-  const { drizzle, useCacheCall, useCacheEvents } = drizzleReactHooks.useDrizzle();
+  const { drizzle, useCacheCall, useCacheEvents, useCacheSend } = drizzleReactHooks.useDrizzle();
   const { web3 } = drizzle;
 
+  const [checkboxesChecked, setCheckboxesChecked] = useState({});
+  const check = (index, event) => {
+    setCheckboxesChecked(old => ({ ...old, [index]: event.target.checked }));
```
Comment (mrice32) [inline voter-dapp/src/ActiveRequests.js]: After thinking about this a bit more, I think this is probably the easiest way to build the "array".

File: voter-dapp/src/ActiveRequests.js (hunk 1 above)
Comment (ptare) [reply voter-dapp/src/ActiveRequests.js]: Done.

File: voter-dapp/src/ActiveRequests.js (hunk 3 above)
Comment (ptare) [reply voter-dapp/src/ActiveRequests.js]: Replaced it with a checkbox that's always disabled.

File: voter-dapp/src/ActiveRequests.js (hunk 4 above)
Comment (ptare) [reply voter-dapp/src/ActiveRequests.js]: Yep, we should. Here, and in one other place in this file.

File: voter-dapp/src/ActiveRequests.js (hunk 2 above)
Comment (ptare) [reply voter-dapp/src/ActiveRequests.js]: Yeah, that definitely seems better. Not immediately obvious how to do that though haha.

Comment (mrice32) [reply voter-dapp/src/ActiveRequests.js]: I say we leave it as is and figure it out later. This also might be useful if the transaction fails for whatever reason and the user wants to try to resubmit with the same boxes checked.

File: voter-dapp/src/ActiveRequests.js (hunk 3 above)
Comment (mrice32) [reply voter-dapp/src/ActiveRequests.js]: SGTM.

**Pull request merged**